runtime/lua/vim/net.lua

```diff
@@ -0,0 +1,71 @@
+local M = {}
+
+--- Makes an HTTP GET request to the given URL.
+---
+--- This function operates in two modes:
+---   - Synchronous (blocking): If `on_response` is omitted, it waits for the
+---   request to complete and returns the result.
+---   - Asynchronous (non-blocking): If `on_response` is provided, it returns
+---   immediately and invokes the callback with the result later.
+---
+--- @param url string The URL for the request.
+--- @param opts? table Optional parameters:
+---   - `verbose` (boolean|nil): Enables curl verbose output.
```

Comment:
(not a blocker) Maybe this should be controlled by the 'verbose' option. Else we will need `verbose` flags on many functions. 

```suggestion
---   - `verbose` (boolean|nil): Enables verbose output.
```

Reply:
```suggestion
---   - `verbose` (boolean|nil): Enables verbose output.
```